packages/backend-for-frontend/src/services/agreementService.ts

```diff
@@ -673,9 +674,43 @@ export async function enrichAgreement(
     consumer.attributes,
     attributes
   );
+
+  const delegations =
+    await clients.delegationProcessClient.delegation.getDelegations({
+      queries: {
+        delegatorIds: [agreement.consumerId],
+        eserviceIds: [agreement.eserviceId],
+        offset: 0,
+        limit: 1,
+      },
+      headers: ctx.headers,
+    });
+
+  const requesterId = ctx.authData.organizationId;
+
+  const delegation = delegations.results.find(
+    (d) => d.delegateId === requesterId || d.delegatorId === requesterId
+  );
```

Comment:
Proposal: add the delegation data whenever the delegation exists, without checking the requester - as you mentioned, having these checks in the BFF is scary given that we don't have any other check on the requester for agreements, at the moment.

We will then add checks on producer/consumer/delegators as assertions on the service as part of the dedicated task tracked here: https://pagopa.atlassian.net/browse/PIN-5849

```suggestion
```

Reply:
I would also comment on the SRS in the part that talks about visibility, linking to the task, to make sure it's tracked there as well:

<img width="455" alt="image" src="https://github.com/user-attachments/assets/c67db4a0-f656-4526-9a98-f0a8ec4f49ae" />

@AsterITA could you do that in case? I think I am not allowed to comment 🤔

Reply:
ok, so the idea is to implement the delegator/delegate check in the other task that should implement the "right" access assertion.
For me it's fine!